### DISCUSSION TITLE
404 "Missing or invalid oauth2 subscription client" when auth provider redirecting user back to pocketbase after authorization

### DISCUSSION THREAD

**Discussion by tobycm** [pocketbase/pocketbase#2566] [Q&A]
I use PocketBase SDK in my flutter app that runs on Windows and Web.
Social auth works fine in the Windows app. However, when I attempt to auth on web, after got redirected back from the auth provider to PocketBase this error is thrown:

```json
{
  "code": 404,
  "message": "Missing or invalid oauth2 subscription client.",
  "data": {}
}
```

**Comment by ganigeorgiev** [comment]
The "all in one" OAuth2 call of the Dart SDK relies on the realtime subscriptions but currently the Dart SDK doesn't support streamed responses on the web platform (see https://github.com/pocketbase/dart-sdk#limitations).

If you want to use the OAuth2 with the Dart SDK and to target browsers, you'll have to implement the "Manual code exchange" flow (_currently we have only a plain JS example in https://pocketbase.io/docs/authentication/#oauth2-integration, but the process should be similar with Dart_).

**Reply by tobycm** [reply]
oh ok, thanks a lot

**Comment by jainmohit** [comment]
Have we got any solution for the same considering we have new version of pocketbase as i do not want to use the manual code exchange was hoping to use "All in one"

**Reply by jainmohit** [reply]
Using PocketBase v0.18.6 version

**Reply by ganigeorgiev** [reply]
@jainmohit If you want to use the "All in one" auth flow in Flutter web you'll have to specify a custom http client implementation. You can find an example in https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#limitations.

In any case, this is an outdated discussion. If the above doesn't help please open a new discussion and elaborate clearly on your use case and what platform you are targeting.
